sdk/master_board_sdk/src/motor.cpp

```diff
@@ -75,15 +77,25 @@ bool Motor::GetIndexToggleBit()
   return index_toggle_bit;
 }
 
-  float Motor::GetPosition()
-  {
-    return position;
-  }
-  float Motor::GetVelocity()
-  {
-    return velocity;
-  }
-  float Motor::GetCurrent()
-  {
-    return current;
-  }
\ No newline at end of file
+float Motor::GetPosition()
+{
+  return direction * (position - position_offset);
+}
+float Motor::GetVelocity()
+{
+  return direction * velocity;
+}
+float Motor::GetCurrent()
+{
+  return direction * current;
+}
+
+void Motor::SetDirection(bool reverse_motor_direction)
+{
+  direction = reverse_motor_direction ? -1 : 1;
+}
+
+void Motor::ZeroPosition()
+{
+  position_offset = position;
```

Comment:
I would prefer that we never use actual state of the motor in the sdk itself.  We don't know how old is the position since it's up to the user code to trigger the reading of the packet. 
I suggest that we instead have a SetOffset function, that will work with absolute offset. The user can always call SetOffset(getPosition()) if needed.

Also for calibration procedure, we will have an absolute offset with respect to the index of the Motor.

Reply:
Good point!

Reply:
I was thinking about using `SetOffset(getPosition())` at first as well. The problem is that `GetPosition()` returns a relative position. What we could do is `SetOffset(GetPosition)`, which works with relative offset and `SetAbsoluteOffset(GetAbsolutePosition)`, which sets the offset based on absolute positions.